toqito/matrices/cyclic_permutation.py

```diff
@@ -0,0 +1,61 @@
+"""Generate a cyclic permutation matrix."""
+
+import numpy as np
+
+
+def cyclic_permutation_matrix(n: int) -> np.ndarray:
+    r"""Create the cyclic permutation matrix for a given dimension :code:`n` :cite:`WikiCyclicPermutation`.
```

Comment:
Might want to make it obvious somewhere in the docstring that this matrix only has 1's and 0's throughout.